rotkehlchen/api/rest.py

```diff
@@ -2172,26 +2172,18 @@ def _do_import_data(
     def _import_data(
             self,
             source: IMPORTABLE_LOCATIONS,
-            filepath: Union[FileStorage, Path],
+            filepath: Path,
             timestamp_format: Optional[str],
     ) -> Dict[str, Any]:
-        if isinstance(filepath, Path):
-            success, msg = self._do_import_data(
-                source=source,
-                filepath=filepath,
-                timestamp_format=timestamp_format,
-            )
-        else:
-            with tempfile.TemporaryDirectory() as temp_directory:
-                filename = filepath.filename if filepath.filename else f'{source}.csv'
-                saved_filepath = Path(temp_directory) / filename
-                filepath.save(str(saved_filepath))
-                success, msg = self._do_import_data(
-                    source=source,
-                    filepath=saved_filepath,
-                    timestamp_format=timestamp_format,
-                )
-
+        """Imports the data depending on the `source` and cleanup the tmp dir created."""
+        success, msg = self._do_import_data(
+            source=source,
+            filepath=filepath,
+            timestamp_format=timestamp_format,
+        )
+        tmp_dir = tempfile.gettempdir()
+        if filepath.is_relative_to(tmp_dir) is True:
+            shutil.rmtree(filepath.parents)
```

Comment:
This will delete the entire filesystem @prettyirrelevant :scream: :sweat: :cry:

Reply:
Ohhhh I get. Thanks for the detailed response. I'll take a look at your implementation